### DISCUSSION TITLE
Set Dashboard filter box value before email report?

### DISCUSSION THREAD

**Discussion by waytoyou** [apache/superset#18308] [Q&A / Help]
In the dashboard  I put filter box to control the data in table. When I sent report via email. The report shows the report by default value in filter box. Is there any solution to email  different report? Thanks a lot.

**Comment by amitmiran137** [comment]
We should leverage filtersets for choosing filters in a reports for scheduling purposes

With that dashboard will load with all the nessecery filters and then screenshot will be taken

**Comment by waytoyou** [comment]
@amitmiran137 Yeah!!! Now report settings seems can not leverage any filtersets.

**Comment by VincentZ112** [comment]
Follow up on this ticket. I am trying to send different customers the same report with different filters. This feature could be really helpful.

**Comment by DonWizu** [comment]
I subscribe to the thread as well. Quite interested on this feature. Having the same issues as we need to send reports with different filters. So far the only solution is to create two dashboards with different default filters and create two different reports, but it's harder to maintain.

**Comment by thymeC** [comment]
How is things going? I would like to sent reports with filter set with different value, too.

**Comment by jitendra-dangi** [comment]
Do we have any solution for this?

**Comment by rusackas** [comment]
Dashboard filters are more of a consumption/end-user experience. They are not, typically, considered part of the dashboard-authoring experience. 

I know this issue is old, so I might need a little fact checking on what I'm about to propose, but a couple of caveats for the thread here:
* The FilterBox component is on the road to deprecation. Native Filters are the path forward.
* Filter Sets are also likely on the path to removal/deprecation. They were experimental, and have not been largely adopted or fleshed out/stabilized.

So... for Dashboard Native Filters, and how to accomplish this, a few key points:
• Adding a filter to a dashboard is a dashboard-authoring event. Others see this when loading the dash.
• Applying values to these filters is an end-user consumption experience, Others (including alerts/reports) do NOT see these values changed.

I _believe_ you can work around this by setting default values when creating the filter. These default filter values are automatically applied to the dashboard on load, and should be picked up for alets and reports. End users can still change them freely.

**Reply by sker65** [reply]
I suspect the proposed workaround is not really applicable as you would need to copy dashs just to set different default filter params. Some postings above it was already mentioned that sending out a dash to different customers with different filter params is a quite common use case I guess. While an interactive filtering may be a user facing UI feature, a "set of parameters" for a dash is quite common. Superset even supports URL params, so it seems quite consequential to also use such a "thing" when rendering dashboards for reports.

**Reply by SeanStock-THG** [reply]
What about adding the permalink key as an option?
I had mentioned this here [(https://github.com/apache/superset/discussions/22905)]

**Comment by pasqualtroncone** [comment]
as @sker65 pointed out, this is quite common request on any BI software. It is called Bursting and is very useful to deliver segmented data to areas / people on your organization. You can archive this with some work around on PowerBI, Tableau, etc. OBI on the other hand has it integrated on his software. Take it from _[Fusion Middleware Data Modeling Guide for Oracle Business Intelligence Publisher](https://docs.oracle.com/middleware/bi12214/bip/BIPDM/GUID-A926D588-426F-47C7-8A9A-B0A7959C5814.htm#GUID-8CDBF7E1-0828-4572-872C-CCDBC922547B)_

> Bursting is a process of splitting data into blocks, generating documents for each block, and delivering the documents to one or more destinations.
>
>The data for the report is generated by executing a query once and then splitting the data based on a Key value. For each block of the data, a separate document is generated and delivered.
>Using Oracle BI Publisher bursting enables splitting a single report based on an element in the data model and deliver the report based on a second element in the data model. Driven by the delivery element, you can apply a different template, output format, delivery method, and locale to each split segment of the report. Example implementations include:
>- Invoice generation and delivery based on customer-specific layouts and delivery preference.
>- Financial reporting to generate a master report of all cost centers, splitting out individual cost center reports to the appropriate manager.
>- Generation of pay slips to all employees based on one extract and delivered through e-mail.
